Fix tests when NO_COLOR is set in the calling environment

mgorny opened the pull request:
Add `FORCE_COLOR` and `NO_COLOR` variables to the `isolate()` fixture to ensure that these two variables do not affect internal test output. This fixes the following two test failures when pytest is called with `NO_COLOR` set:

```
FAILED tests/unit/test_exceptions.py::TestDiagnosticPipErrorPresentation_ASCII::test_complete_color - AssertionError: assert '\x1b[1merror...ing harder.\n' == '\x1b[1;31mer...ing harder.\n'
FAILED tests/unit/test_exceptions.py::TestDiagnosticPipErrorPresentation_Unicode::test_complete_color - AssertionError: assert '\x1b[1merror...ing harder.\n' == '\x1b[1;31mer...ing harder.\n'
```

webknjaz reviewed (commented):
Cool! This is something I was going to suggest at some point.

To make this change truly tested in pip's own CI, how about including a commit reverting https://github.com/pypa/pip/commit/1b33f4b944b46f9d8bd6380201ef1e6c33fd944a?

webknjaz commented:
_(context comment)_

I recently discovered that there are bugs related with `FORCE_COLOR` and `NO_COLOR`, coming from different places in pip and the underlying Rich library: https://github.com/pypa/pip/issues/12405.

As of today, using both `FORCE_COLOR` and `NO_COLOR` forces pip to use ANSI-escapes that make parts of output **bold**, while keeping it *monochrome*. And this breaks tools that read piped output.
I even started discussions on the trackers of the corresponding standards: https://github.com/jcs/no_color/issues/267 / https://github.com/donatj/force-color.org/issues/8. Rich hasn't recognized this behavior as a bug, so far.

mgorny commented:
> To make this change truly tested in pip's own CI, how about including a commit reverting [1b33f4b](https://github.com/pypa/pip/commit/1b33f4b944b46f9d8bd6380201ef1e6c33fd944a)?

I'm sorry about forgetting about this pull request. I've added the revert now and rebased.

mgorny commented:
Ok, so the problem with `tests/functional/test_inspect.py::test_inspect_basic` is that `simple_script` fixture is initiated "outside" `isolate` fixture, and effectively copies `os.environ` prior to the variables being stripped. How would you prefer that being fixed?

The options I can think of are:
1. Reducing `simple_script` to function-scope fixture (it's used by one test anyway).
2. Modifying `PipTestEnvironment` to unconditionally set `NO_COLOR` for executed scripts, or to unset `FORCE_COLOR`.

webknjaz commented:
(1) sounds reasonable, if you ask me.

mgorny commented:
Ok, I think it's good now.

webknjaz commented:
@pradyunsg I think this should be mergeable

mgorny commented:
Thank you!
